conda_forge_tick/auto_tick.py

```diff
@@ -477,6 +484,74 @@ def migration_factory(
         )
 
 
+def _outside_pin_range(pin_spec, current_pin, new_version):
+    pin_level = len(pin_spec.split('.'))
+    current_split = current_pin.split('.')
+    new_split = new_version.split('.')
+    # if our pin spec is more exact than our current pin then rebuild more precisely
+    if pin_level > len(current_split):
+        return True
+    for i in range(pin_level):
+        if current_split[i] != new_split[i]:
+            return True
+    return False
+
+
+def create_migration_yaml_creator(migrators: MutableSequence[Migrator], gx: nx.DiGraph):
+    with indir("../conda-forge-pinning-feedstock/recipe"):
+        pinnings = parse_config_file('conda_build_config.yaml',
+                                     config=Config(**CB_CONFIG))
+    feedstocks_to_be_repinned = []
+    for k, package_pin_list in pinnings.items():
+        # exclude non-package keys
+        if k not in gx.nodes and k not in gx.graph['outputs_lut']:
+            # conda_build_config.yaml can't have `-` unlike our package names
+            k = k.replace('_', '-')
+        # replace sub-packages with their feedstock names
+        k = gx.graph['outputs_lut'].get(k, k)
+
+        if (k in gx.nodes) and not gx.nodes[k]['payload'].get('archived', False) and gx.nodes[k]['payload'].get('version') and k not in feedstocks_to_be_repinned:
+
+            current_pins = list(map(str, package_pin_list))
+            current_version = str(gx.nodes[k]['payload']['version'])
+
+            # XXX: uncomment following line and remove the one after once graph
+            # has been re-loaded with the pins properly
+
+            # meta_yaml = gx.nodes[k]['payload']['meta_yaml']
+            meta_yaml = parse_meta_yaml(gx.nodes[k]['payload']['raw_meta_yaml'])
+
+            # find the most stringent max pin for this feedstock if any
+            pin_spec = ''
+            for block in [meta_yaml] + meta_yaml.get("outputs", []) or []:
+                build = block.get('build', {}) or {}
+                # get the most stringent pin spec from the recipe block
+                max_pin = max(build.get('run_exports', ['']), key=len)
+                if len(max_pin) > len(pin_spec):
+                    pin_spec = max_pin
```

Comment:
We need to check the name of the run export here.

Reply:
Why? If it's a run export in that feedstock doesn't it apply to that feedstock?

Reply:
For feedstocks with more than one output, the run export can be for different outputs and they may be pinned at different levels.

Reply:
Run exports can be to a different package. For example `gcc_linux-64` exports `libgcc-ng` and 2 of them are in 2 feedstocks.

Reply:
So should we be opening migrations that state the package name rather than the feedstock? Can the current migrator handle that? (mostly I'm asking if I need to go in fix that bit)

Reply:
Huh? I don't follow the question. My comment here is simply that to get the right pinning level you need to be sure the run export is from the correct package.

Reply:
Can `run_exports` pin a package not within its feedstock?

Reply:
yes